llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -1836,6 +1841,7 @@ class VPWidenIntOrFpInductionRecipe : public VPWidenInductionRecipe {
                                Step, IndDesc, DL),
         Trunc(nullptr) {
     addOperand(VF);
+    addOperand(VF); // Dummy StepVector replaced in convertToConcreteRecipes
```

Comment:
do we need the dummy operands or can we add the operand when needed?

Reply:
I tried adding the operands at the call site this again and it ends up needing a bunch more code because we have to create the step vector in both createWidenInductionRecipes and tryToConvertVPInstructionsToVPRecipes, and also we need to recreate it again in optimizeVectorInductionWidthForTCAndVFUF if the bitwidth is reduced. 

It also doesn't really work well when the operand is added in `convertToConcreteRecipes` because then getLastUnrolledPartOperand` and `getSplatVFValue` get confused with the number of operands in `isUnrolled` . 

The dummy operand would hopefully only be temporary and would go away with https://github.com/llvm/llvm-project/pull/118638 whenever the recipe is fully expanded.

Reply:
Oh good idea, I've changed it to poison in e73e9944eedd280f9b3b2f959affc61425aa6e59